thunder/transforms/materialization.py

```diff
@@ -55,3 +83,69 @@ def module_init_from_transformed_state_dict(transform: MaterializationTransform,
             model.load_state_dict(state_dict)
 
         return module_init_from_transformed_state_dict
+
+    @staticmethod
+    def init_from_original_module_init():
```

Comment:
question kind of orthogonal to this pr: would it be possible that one passes sharded state dict to this one?

Reply:
If you have a sharded state dict, you should be able to use `module_init_from_transformed_state_dict` (provided you don't have other state-dict-changing transformations afterwards.